iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -534,10 +565,13 @@ public boolean canProvideColStatistics(org.apache.hadoop.hive.ql.metadata.Table
   }
 
   private boolean canProvideColStats(Table table, long snapshotId) {
-    Path statsPath = getColStatsPath(table, snapshotId);
+    Optional<Path> statsPath = IcebergTableUtil.getColStatsPath(table, snapshotId);
+    if (!statsPath.isPresent()) {
+      return false;
+    }
     try {
-      FileSystem fs = statsPath.getFileSystem(conf);
-      return  fs.exists(statsPath);
+      FileSystem fs = statsPath.get().getFileSystem(conf);
```

Comment:
Do we still need the check filesystem path to confirm the stats file?

I think the result of `IcebergTableUtil::getColStatsPath` is enough.

Reply:
nice catch!